tests/subset.rs

```diff
@@ -56,38 +71,52 @@ fn test_subset<A: Adversary<Subset<NodeId, u8>>>(
             assert_eq!(&inputs[id], value);
         }
 
-        assert_eq!(outputs.iter().cloned().collect::<BTreeSet<_>>(), observer);
+        assert_eq!(
+            outputs.iter().cloned().collect::<BTreeSet<_>>(),
+            expected_value
+        );
     }
 }
 
 fn new_network<A, F>(
     good_num: usize,
     bad_num: usize,
     adversary: F,
-) -> TestNetwork<A, Subset<NodeId, u8>>
+) -> VirtualNet<Subset<NodeId, u8>, A>
 where
     A: Adversary<Subset<NodeId, u8>>,
-    F: Fn(BTreeMap<NodeId, Arc<NetworkInfo<NodeId>>>) -> A,
+    F: Fn() -> A,
 {
     // This returns an error in all but the first test.
     let _ = env_logger::try_init();
 
-    let new_subset =
-        |netinfo: Arc<NetworkInfo<NodeId>>| Subset::new(netinfo, 0).expect("new Subset instance");
-    TestNetwork::new(good_num, bad_num, adversary, new_subset)
+    let mut rng = rand::thread_rng();
```

Comment:
Likewise, please don't use `thread_rng` but use `TestRng` instead.

Reply:
Switched to TestRng